datalad/support/gitrepo.py

```diff
@@ -3444,6 +3466,63 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
         # TODO yield result for commit, prev helper checked hexsha pre
         # and post...
 
+    def _check_for_win_compat(self, files, config):
+        """Check file names for illegal characters or reserved names on Windows
+
+        In the case that a non-Windows-compatible file is detected, warn users
+        about potential interoperability issues.
+
+        Parameters
+        ----------
+        files
+          list of files to add
+        config
+          value of self.config.obtain("datalad.save.windows-compat-warning"),
+          used to choose appropriate behavior. "none" performs no check,
+          "warning" warns in case of incompatibilities, and "error" results in
+          an error result in case of incompatibilities
+        """
+        # don't perform any check when the configuration is set to 'none'
+        if config == 'none':
+            return files, None
```

Comment:
just for annoying me, please add some test run for the `none` so we ensure that there is no side effect and everything functions as expected?

Reply:
that makes sense, sure :)__